code/modules/detective_work/scanner.dm

```diff
@@ -154,61 +162,59 @@
-			if(fingerprints && fingerprints.len)
-				sleep(30)
-				add_log("<span class='info'><B>Prints:</B></span>")
-				for(var/finger in fingerprints)
-					add_log("[finger]")
-				found_something = 1
-
-			// Blood
-			if(blood && blood.len)
-				sleep(30)
-				add_log("<span class='info'><B>Blood:</B></span>")
-				found_something = 1
-				for(var/B in blood)
-					add_log("Type: <font color='red'>[blood[B]]</font> DNA: <font color='red'>[B]</font>")
-
-			//Fibers
-			if(fibers && fibers.len)
-				sleep(30)
-				add_log("<span class='info'><B>Fibers:</B></span>")
-				for(var/fiber in fibers)
-					add_log("[fiber]")
-				found_something = 1
-
-			//Reagents
-			if(reagents && reagents.len)
-				sleep(30)
-				add_log("<span class='info'><B>Reagents:</B></span>")
-				for(var/R in reagents)
-					add_log("Reagent: <font color='red'>[R]</font> Volume: <font color='red'>[reagents[R]]</font>")
-				found_something = 1
-
-			// Get a new user
-			var/mob/holder = null
-			if(ismob(src.loc))
-				holder = src.loc
-
-			if(!found_something)
-				add_log("<I># No forensic traces found #</I>", 0) // Don't display this to the holder user
-				if(holder)
-					to_chat(holder, "<span class='notice'>Unable to locate any fingerprints, materials, fibers, or blood on \the [target_name]!</span>")
-			else
-				if(holder)
-					to_chat(holder, "<span class='notice'>You finish scanning \the [target_name].</span>")
-
-			add_log("---------------------------------------------------------", 0)
-			scanning = 0
-			return
+
+		// We gathered everything. Slowly display the results to the holder of the scanner.
+		var/found_something = 0
+		add_log("<B>[station_time_timestamp()][get_timestamp()] - [target_name]</B>", 0)
+
+		// Fingerprints
+		if(length(fingerprints))
+			sleep(30)
+			add_log("<span class='info'><B>Prints:</B></span>")
+			for(var/finger in fingerprints)
+				add_log("[finger]")
+			found_something = 1
+
+		// Blood
+		if(length(blood))
+			sleep(30)
+			add_log("<span class='info'><B>Blood:</B></span>")
+			found_something = 1
+			for(var/B in blood)
+				add_log("Type: <font color='red'>[blood[B]]</font> DNA: <font color='red'>[B]</font>")
+
+		//Fibers
+		if(length(fibers))
+			sleep(30)
+			add_log("<span class='info'><B>Fibers:</B></span>")
+			for(var/fiber in fibers)
+				add_log("[fiber]")
+			found_something = 1
+
+		//Reagents
+		if(length(reagents))
+			sleep(30)
+			add_log("<span class='info'><B>Reagents:</B></span>")
+			for(var/R in reagents)
+				add_log("Reagent: <font color='red'>[R]</font> Volume: <font color='red'>[reagents[R]]</font>")
+			found_something = 1
+
+		// Get a new user
+		var/mob/holder = null
+		if(ismob(src.loc))
+			holder = src.loc
+
+		if(!found_something)
+			add_log("<I># No forensic traces found #</I>", 0) // Don't display this to the holder user
+			if(holder)
+				to_chat(holder, "<span class='notice'>Unable to locate any fingerprints, materials, fibers, or blood on \the [target_name]!</span>")
```

Comment:
`\the` doesn't do anything here I believe. Need to test this though

Reply:
It seems like it was needed there, since `target_name` was the `name` of the atom, rather than the atom itself.
I've changed it to `[A]`, so it's not needed anymore.